packages/drivers/odsp-driver/src/odspUtils.ts

```diff
@@ -473,3 +474,14 @@ export async function measureP<T>(callback: () => Promise<T>): Promise<[T, numbe
 export function getJoinSessionCacheKey(odspResolvedUrl: IOdspResolvedUrl) {
 	return `${odspResolvedUrl.hashedDocumentId}/joinsession`;
 }
+
+/**
+ * Utility API to check if the type of snapshot is `ISnapshotContents`.
+ * @alpha
+ * @param obj - obj whose type needs to be identified.
+ */
+export function isFullSnapshot(
+	obj: ISnapshotContents | IPartialSnapshotWithContents,
+): obj is ISnapshotContents {
+	return typeof obj === "object" && obj !== null && !("partialSnapshot" in obj);
```

Comment:
Given that you removed undefined & null from argument type, there is no need to check for typeof obj === "object" && obj !== null. Maybe it's good assert (that we actually got an object here), but if you remove it, and it's not an object, it will throw, which feels like better option if garbage is in, then returning false.